mongodb/src/test/java/com/hazelcast/jet/contrib/mongodb/MongoDBContainer.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.contrib.mongodb;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import org.testcontainers.containers.ContainerLaunchException;
+import org.testcontainers.containers.GenericContainer;
+
+import java.util.Collections;
+import java.util.Set;
+
+public class MongoDBContainer<SELF extends MongoDBContainer<SELF>> extends GenericContainer<SELF> {
```

Comment:
I think this self type here is not really necessary, because we won't be extending the MongoDBContainer further (right?) and since it's a very ugly, hard to understand thing, we should probably drop it and just use `MongoDBContainer extends GenericContainer<MongoDBContainer>`.